problem1-twosum/twoSum.py

```diff
@@ -18,19 +18,28 @@ class Solution:
     def __init__(self):
         pass
 
-    def twoSum(self, nums: List[int], target: int) -> List[int]:
+    def tashizanKun(self, nums: List[int], target: int) -> List[int]:
         # Write your solution here
-        return [0, 0]
+        ans_list = []
+        for idx in range(len(nums)):
```

Comment:
Good Solution!! Looks alright to me.

But point of this problem is to think of what is optimal solution.

Right now in the solution you use 2 "for loops". 
That means if length of list is 4. it would run maximum 6 times if the last two number represent solution.
For example [2, 7, 11, 15] target = 26.
loop index (0, 1), (0, 2), (0, 3), (1, 2), (1, 3), (2, 3) -> this is solution sequence.

But ideally we want to avoid more computation. so we should think better Data Structure.

**HINT: Think using dict**

Reply:
@prasad-marne -san, Thank you for review my code!

> Right now in the solution you use 2 "for loops".
this is exactly what I want to avoid, but I couldn't. 
I'll fix the code with using dict.

Reply:
HINT:  
we know:  num1 + num2 = target
How about num2 = target - num1 ?
now you can just check if num2 is present and its position.

Reply:
Algorithm:
        d = {}
        for i, n in enumerate(nums):
            m = target - n
            if m in d:
                return [d[m], i]
            else:
                d[n] = i

Reply:
It makes sense.
I had another problem.
I had a list that is same combination but different order.
ex)
[1,0] [0,1]
this integers in lists indicates index of dict. so they are different  but same output in two sum.
to avoid it, I use copy of dict and del.
but I was not willing to do it, because it looks very complicated.

Reply:
i see. i understand your feeling completely..lol